os/kernel/binary_manager/binary_manager_data.c

```diff
@@ -198,18 +198,24 @@ void binary_manager_add_binlist(FAR struct tcb_s *tcb)
 {
 	int bin_idx;
 
-	if (tcb->group) {
-		bin_idx = tcb->group->tg_binidx;
-		if (bin_idx > 0) {
-			/* Add a tcb to a head of list */
-			tcb->bin_blink = NULL;
-			if (tcb->sched_priority > BM_PRIORITY_MAX) {
-				tcb->bin_flink = BIN_RTLIST(bin_idx);
-				BIN_RTLIST(bin_idx) = tcb;
-			} else {
-				tcb->bin_flink = BIN_NRTLIST(bin_idx);
-				BIN_NRTLIST(bin_idx) = tcb;
+	bin_idx = tcb->group->tg_binidx;
+
+	/* A index of binary, bin_idx is greater than 0 only if tcb is a thread of user binary */
+	if (bin_idx > 0) {
+		/* Add a tcb to a head of list */
+		tcb->bin_blink = NULL;
+		if (tcb->sched_priority > BM_PRIORITY_MAX) {
+			tcb->bin_flink = BIN_RTLIST(bin_idx);
```

Comment:
When BIN_RTLIST is null, this line is unnecessary. Let's move into if case. But then, you should change the condition to check BIN_RTLIST.